docs/apache-airflow/database-erd-ref.rst

```diff
@@ -0,0 +1,23 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+ERD Schema of the Database
+''''''''''''''''''''''''''
+
+Here is the current Database schema diagram.
```

Comment:
Should we note that DB schema considered internal and can change between minor/feature versions?
I've seen cases of ETLs that query Airflow DB directly :/

Reply:
Yeah -- for this reason alone I'd rather we _DONT_ include an ERD in published docs on `airflow.apache.org`.

Reply:
(My view here is not blocking, but I do want a "this is internal" notice if we keep it)

Reply:
Yeah, instead of in the public docs, maybe we should toss it in the `dev` folder instead?

Reply:
I thought the same: `dev` initially. That would be "good enough" for many cases.

But then I looked at our docs and - we already have db migrations in public docs. Somehow we already decided that we need to make this part "public" and make our users aware of that (and we have the CLI commands. And somehow, those migrations require the users to know "A little" about our database - to know what state it is in and understand informationa about those tables changed. 

We also do some of the troubleshooting that is coming (I added it recently as a desperate attempt of having something that I can direct poor MySQL users to) - with more of the information about handling - mainly MySQL - migration problems: https://github.com/apache/airflow/blob/main/docs/apache-airflow/installation/upgrading.rst#handling-migration-problems

So this "Airlfow DB is internal" abstraction is already leaky like crazy :). 

I think adding this page in the public docs is quite natural consequence of those and is better than not publishing it (because of all the other information we already publish), but indeed adding a disclaimer that this DB structure  is internal and might change at any time, is probably best.

WDYT?

Reply:
The DB is part of Airflow deployment so users should be aware of it and to be able to understand migrations. However that is very narrow window of before/during migration and even so the issues are most likely about specific problems where the **data context is irrelevant** it's just about column type / index etc. So you don't care what the data/structure represent you just care about finishing migrations with no issues.

When a user is asking to see ERD diagram his intention is to **study** the application and possibly make some integration to the DB **directly**.  The data context is interesting for this. You really care about what the structure represent. This is OK when you want to be a developer of application (like contributing to Airflow in our case) but wrong for other cases.

Regardless of if ERD will be on website or internal in /dev we should have a website entry that explain it's risky to rely directly on the DB tables.
We can also have this explanation and mention that if for some reason someone needed the ERD for a valid use case that we didn't think about they can find it in the repo via the contribution guide. I think this is a good balance between not "inviting abuse" yet not make it really hard to find for people who understand the risk and choose to take it.

Reply:
If the correct answer is `dev`, we shouldn't be reluctant about putting this there just because there are other internal topics already covered in the docs. We can always move things from the docs into `dev` if we need to. Given that the database is not a user-facing resource - and ideally it would not be an admin-facing one - I personally think it belongs in `dev`. It keeps the docs more focused on typical users' needs.

But I don't feel strongly. In fact, there are arguments for having it in the docs - versioning, for one. If this does end up it in the docs, perhaps we could create an Internals section? That could help transmit the right expectation.

Reply:
This is what I addressed In my warning proposals. And yes - I put exactly in the warning that the ERD diagram is really needed in this "narrow migration" window and should not be accessed directly. I think the intention of seeing the ERD diagram is also about the migration. When the migration is happening and you see "this column in this table is about to be added", you want to likely assess how big the table is and what will happen (when you are proffessional DB admin - and this ERD diagram is for those people). Those peopla are not likely we versed in the SQLAlchemy/Python code modelling the DB, but they will perfectly understand what it means when a foreign key and new column is going to be added to their table. And yeah - they likely can generate the ERD diagram using whatever tools they have at hand, but At least they will be able to see if their table is as expect before the migration - or in case they have any migraiton problems, to compare what "airflow" expects and what they "see". 

I think pretty much all our DB-related explanations are around migration and it should be the same for the ERD -diagram. This is mostly a migration helper. And I tried to explain this in the warning.

Reply:
> The main purpose of this diagram is to help with troubleshooting and understanding of the
>    internal Airflow DB architecture in case you have any problems with the database - for example
>    when dealing with problems with migrations. See also :doc:`migrations-ref` for
>    list of detailed database migrations that are applied when running migration script and
>    `db command <cli-and-env-variables-ref.html#db>`_ for the commands that you can use to manage
>    the migrations.

Reply:
> But I don't feel strongly. In fact, there are arguments for having it in the docs - versioning, for one. If this does end up it in the docs, perhaps we could create an Internals section? That could help transmit the right expectation.

I like the idea - how about we put it to  separate section in the docs Both "migrations" and "ERD" are now in "References" section. How about we move both to "Internal DB details" section below "References" @eladkal ?

Reply:
Just to comment on motivation - I do agree that exposing it too much and making people think this is part of "API" of Airflow is a bad thing - I just think it is sometimes useful by our users, so it's good to have it in the user's docs, we just need to find the right way to emphasise those are really internals tha should not be used beyond the migration time where it might be useful to solve some problems that users might find easier to solve by having such diagram handy.